File: frontend/src/model/Appointment.js
class Appointment {
    id
    jobApplication
    appointmentDate
    isChosen

    constructor(){
        this.reset()
    }

    reset(){
        this.id = 0
        this.jobApplication = null
        this.appointmentDate = ""
        this.isChosen = false
    }

    init(appointment){
        if(appointment.id) this.id = appointment.id
        else delete this.id
        if(appointment.jobApplication) this.jobApplication = appointment.jobApplication
        else delete this.jobApplication
        if(appointment.appointmentDate) this.appointmentDate = appointment.appointmentDate
        else delete this.appointmentDate
        if(appointment.isChosen) this.isChosen = appointment.isChosen
        else delete this.isChosen
    }

}

export default Appointment

File: src/main/java/cal/projeteq3/glucose/model/Appointment.java
package cal.projeteq3.glucose.model;

import cal.projeteq3.glucose.model.jobOffer.JobApplication;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.ToString;

import java.time.LocalDateTime;

@Entity
@Data
@Builder
@AllArgsConstructor
public class Appointment {
    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne
    @ToString.Exclude
    private JobApplication jobApplication;

    private LocalDateTime appointmentDate;

    private boolean isChosen;

    public Appointment(){
        this.isChosen = false;
    }
}
